# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/process_mitigations_win32k_interception.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/process_mitigations_win32k_interception.cc
--- a/security/sandbox/chromium/sandbox/win/src/process_mitigations_win32k_interception.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/process_mitigations_win32k_interception.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -56,25 +56,48 @@
           (unicode_string->Length / sizeof(unicode_string->Buffer[0])));
 }
 
-}  // namespace
+bool CallMonitorInfo(HMONITOR monitor, MONITORINFOEXW* monitor_info_ptr) {
+  // We don't trust that the IPC can work this early.
+  if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
+    return false;
+
+  void* ipc_memory = GetGlobalIPCMemory();
+  if (!ipc_memory)
+    return false;
 
-BOOL WINAPI TargetGdiDllInitialize(
-    GdiDllInitializeFunction orig_gdi_dll_initialize,
-    HANDLE dll,
-    DWORD reason) {
-  return TRUE;
+  CrossCallReturn answer = {};
+  SharedMemIPCClient ipc(ipc_memory);
+  InOutCountedBuffer buffer(monitor_info_ptr, sizeof(*monitor_info_ptr));
+  ResultCode code = CrossCall(ipc, IPC_USER_GETMONITORINFO_TAG,
+                              static_cast<void*>(monitor), buffer, &answer);
+
+  if (code != SBOX_ALL_OK)
+    return false;
+
+  if (answer.win32_result != ERROR_SUCCESS)
+    return false;
+
+  return true;
 }
 
-HGDIOBJ WINAPI TargetGetStockObject(
-    GetStockObjectFunction orig_get_stock_object,
-    int object) {
-  return reinterpret_cast<HGDIOBJ>(NULL);
+}  // namespace
+
+BOOL WINAPI
+TargetGdiDllInitialize(GdiDllInitializeFunction orig_gdi_dll_initialize,
+                       HANDLE dll,
+                       DWORD reason) {
+  return true;
 }
 
-ATOM WINAPI TargetRegisterClassW(
-    RegisterClassWFunction orig_register_class_function,
-    const WNDCLASS* wnd_class) {
-  return TRUE;
+HGDIOBJ WINAPI
+TargetGetStockObject(GetStockObjectFunction orig_get_stock_object, int object) {
+  return nullptr;
+}
+
+ATOM WINAPI
+TargetRegisterClassW(RegisterClassWFunction orig_register_class_function,
+                     const WNDCLASS* wnd_class) {
+  return true;
 }
 
 BOOL WINAPI TargetEnumDisplayMonitors(EnumDisplayMonitorsFunction,
@@ -83,16 +106,16 @@
                                       MONITORENUMPROC lpfnEnum,
                                       LPARAM dwData) {
   if (!lpfnEnum || hdc || lprcClip) {
-    return FALSE;
+    return false;
   }
 
   // We don't trust that the IPC can work this early.
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
-    return FALSE;
+    return false;
 
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
-    return FALSE;
+  if (!ipc_memory)
+    return false;
 
   CrossCallReturn answer = {0};
   answer.nt_status = 0;
@@ -102,27 +125,24 @@
   ResultCode code =
       CrossCall(ipc, IPC_USER_ENUMDISPLAYMONITORS_TAG, result_buffer, &answer);
 
-  if (code != SBOX_ALL_OK) {
-    return FALSE;
-  }
+  if (code != SBOX_ALL_OK)
+    return false;
 
-  if (answer.win32_result) {
-    return FALSE;
-  }
+  if (answer.win32_result)
+    return false;
 
-  if (result.monitor_count > kMaxEnumMonitors) {
-    return FALSE;
-  }
+  if (result.monitor_count > kMaxEnumMonitors)
+    return false;
 
   for (uint32_t monitor_pos = 0; monitor_pos < result.monitor_count;
        ++monitor_pos) {
-    BOOL continue_enum =
+    bool continue_enum =
         lpfnEnum(result.monitors[monitor_pos], nullptr, nullptr, dwData);
     if (!continue_enum)
-      return FALSE;
+      return false;
   }
 
-  return TRUE;
+  return true;
 }
 
 BOOL WINAPI TargetEnumDisplayDevicesA(EnumDisplayDevicesAFunction,
@@ -130,49 +150,22 @@
                                       DWORD iDevNum,
                                       PDISPLAY_DEVICEA lpDisplayDevice,
                                       DWORD dwFlags) {
-  return FALSE;
-}
-
-static BOOL CallMonitorInfo(HMONITOR monitor,
-                            MONITORINFOEXW* monitor_info_ptr) {
-  // We don't trust that the IPC can work this early.
-  if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
-    return FALSE;
-
-  void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
-    return FALSE;
-
-  CrossCallReturn answer = {};
-  SharedMemIPCClient ipc(ipc_memory);
-  InOutCountedBuffer buffer(monitor_info_ptr, sizeof(*monitor_info_ptr));
-  ResultCode code = CrossCall(ipc, IPC_USER_GETMONITORINFO_TAG,
-                              static_cast<void*>(monitor), buffer, &answer);
-
-  if (code != SBOX_ALL_OK) {
-    return FALSE;
-  }
-
-  if (answer.win32_result != ERROR_SUCCESS)
-    return FALSE;
-
-  return TRUE;
+  return false;
 }
 
 BOOL WINAPI TargetGetMonitorInfoA(GetMonitorInfoAFunction,
                                   HMONITOR monitor,
                                   MONITORINFO* monitor_info_ptr) {
   if (!monitor_info_ptr)
-    return FALSE;
+    return false;
   DWORD size = monitor_info_ptr->cbSize;
-  if (size != sizeof(MONITORINFO) && size != sizeof(MONITORINFOEXA)) {
-    return FALSE;
-  }
+  if (size != sizeof(MONITORINFO) && size != sizeof(MONITORINFOEXA))
+    return false;
   MONITORINFOEXW monitor_info_tmp = {};
   monitor_info_tmp.cbSize = sizeof(monitor_info_tmp);
-  BOOL success = CallMonitorInfo(monitor, &monitor_info_tmp);
+  bool success = CallMonitorInfo(monitor, &monitor_info_tmp);
   if (!success)
-    return FALSE;
+    return false;
   memcpy(monitor_info_ptr, &monitor_info_tmp, sizeof(*monitor_info_ptr));
   if (size == sizeof(MONITORINFOEXA)) {
     MONITORINFOEXA* monitor_info_exa =
@@ -181,28 +174,26 @@
                                monitor_info_exa->szDevice,
                                sizeof(monitor_info_exa->szDevice), nullptr,
                                nullptr)) {
-      return FALSE;
+      return false;
     }
   }
-  return TRUE;
+  return true;
 }
 
 BOOL WINAPI TargetGetMonitorInfoW(GetMonitorInfoWFunction,
                                   HMONITOR monitor,
                                   LPMONITORINFO monitor_info_ptr) {
   if (!monitor_info_ptr)
-    return FALSE;
+    return false;
   DWORD size = monitor_info_ptr->cbSize;
-  if (size != sizeof(MONITORINFO) && size != sizeof(MONITORINFOEXW)) {
-    return FALSE;
-  }
+  if (size != sizeof(MONITORINFO) && size != sizeof(MONITORINFOEXW))
+    return false;
   MONITORINFOEXW monitor_info_tmp = {};
   monitor_info_tmp.cbSize = sizeof(monitor_info_tmp);
-  BOOL success = CallMonitorInfo(monitor, &monitor_info_tmp);
-  if (!success)
-    return FALSE;
+  if (!CallMonitorInfo(monitor, &monitor_info_tmp))
+    return false;
   memcpy(monitor_info_ptr, &monitor_info_tmp, size);
-  return TRUE;
+  return true;
 }
 
 static NTSTATUS GetCertificateCommon(
@@ -221,7 +212,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   ScopedSharedMemory buffer(certificate_size);
@@ -274,7 +265,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   CrossCallReturn answer = {};
@@ -334,7 +325,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   CrossCallReturn answer = {};
@@ -342,9 +333,8 @@
   ResultCode code = CrossCall(ipc, IPC_GDI_DESTROYOPMPROTECTEDOUTPUT_TAG,
                               static_cast<void*>(protected_output), &answer);
 
-  if (code != SBOX_ALL_OK) {
+  if (code != SBOX_ALL_OK)
     return STATUS_ACCESS_DENIED;
-  }
 
   return answer.nt_status;
 }
@@ -362,7 +352,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   ScopedSharedMemory buffer(sizeof(*parameters));
@@ -393,7 +383,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   ScopedSharedMemory buffer(base::checked_cast<uint32_t>(max_size));
@@ -424,7 +414,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   CrossCallReturn answer = {};
@@ -447,7 +437,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   CrossCallReturn answer = {};
@@ -474,7 +464,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   DXGKMDT_OPM_ENCRYPTED_PARAMETERS temp_parameters = *parameters;
@@ -505,7 +495,7 @@
   if (!SandboxFactory::GetTargetServices()->GetState()->InitCalled())
     return STATUS_ACCESS_DENIED;
   void* ipc_memory = GetGlobalIPCMemory();
-  if (ipc_memory == NULL)
+  if (!ipc_memory)
     return STATUS_ACCESS_DENIED;
 
   CrossCallReturn answer = {};
@@ -531,4 +521,3 @@
 }
 
 }  // namespace sandbox
-